compendium/DeclarativeServices/src/metadata/ComponentMetadata.hpp

```diff
@@ -56,6 +56,10 @@ struct ComponentMetadata
   std::vector<ReferenceMetadata> refsMetadata;
   ServiceMetadata serviceMetadata;
   std::unordered_map<std::string, cppmicroservices::Any> properties;
+  std::string configurationPolicy;
+  std::vector<std::string> configurationPids;
+  std::string factory;
```

Comment:
Jeff: change variable names to something more descriptive like factorycomponentID and factorycomponentproperties